### PR TITLE
[SPARK-33082][SPARK-20202][BUILD][SQL][FOLLOW-UP] Remove Hive 1.2 workarounds and Hive 1.2 profile in Jenkins script

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR removes the leftover of Hive 1.2 workarounds and Hive 1.2 profile in Jenkins script.

- `test-hive1.2` title is not used anymore in Jenkins
- Remove some comments related to Hive 1.2
- Remove unused codes in `OrcFilters.scala` @ Hive
- Test `spark.sql.hive.convertMetastoreOrc` disabled case for the tests added at SPARK-19809 and SPARK-22267

### Why are the changes needed?

To remove unused codes & improve test coverage

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

Manually ran the unit tests. Also It will be tested in CI in this PR.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @dongjoon-hyun FYI

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFilters.scala [hunk 1]
```diff
@@ -64,185 +64,7 @@ import org.apache.spark.sql.types._
...
-
-      case In(attribute, values) if isSearchableType(dataTypeMap(attribute)) =>
-        val bd = builder.startAnd()
-        val method = findMethod(bd.getClass, "in", classOf[String], classOf[Array[Object]])
-        Some(method.invoke(bd, attribute, values.map(_.asInstanceOf[AnyRef]))
-          .asInstanceOf[Builder].end())
-
-      case _ => None
-    }
+    DatasourceOrcFilters.createFilter(schema, filters)
```
Comment (HyukjinKwon) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFilters.scala:68]: It seems not used anymore after https://github.com/apache/spark/commit/a127387a53e1a24e76de83c5a1858fcdbd38c3a2#diff-6cac9bc2656e3782b0312dceb8c55d47L77-L88

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcFilterSuite.scala [hunk 2]
```diff
@@ -296,11 +295,6 @@ class HiveOrcFilterSuite extends OrcTest with TestHiveSingleton {
 
   test("filter pushdown - combinations with logical operators") {
     withOrcDataFrame((1 to 4).map(i => Tuple1(Option(i)))) { implicit df =>
-      // Because `ExpressionTree` is not accessible at Hive 1.2.x, this should be checked
-      // in string form in order to check filter creation including logical operators
-      // such as `and`, `or` or `not`. So, this function uses `SearchArgument.toString()`
-      // to produce string expression and then compare it to given string expression below.
-      // This might have to be changed after Hive version is upgraded.
```
Comment (HyukjinKwon) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcFilterSuite.scala:303]: Even if we can access `ExpressionTree` from `SearchArgument`, we can't construct them to compare to use in tests. This seems what Hive does as well, see https://github.com/apache/hive/blob/master/ql/src/test/org/apache/hadoop/hive/ql/io/sarg/TestSearchArgumentImpl.java

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFileFormat.scala [hunk 3]
```diff
@@ -296,7 +296,7 @@ private[orc] class OrcOutputWriter(
 
   override def close(): Unit = {
     if (recordWriterInstantiated) {
-      // Hive 1.2.1 ORC initializes its private `writer` field at the first write.
+      // Hive ORC initializes its private `writer` field at the first write.
```
Comment (HyukjinKwon) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFileFormat.scala:299]: This is still same in the master of Hive.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcFilterSuite.scala (hunk 2 above)
Comment (HyukjinKwon) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcFilterSuite.scala:303]: BTW, I wrote this comment. I am sure we can remove it :-).

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFilters.scala (hunk 1 above)
Comment (wangyum) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFilters.scala:68]: Could we remove this file? We can use `org.apache.spark.sql.execution.datasources.orc.OrcFilters` directly.

**Comment by HyukjinKwon** [comment]
retest this please

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you so much for the follow-up, @HyukjinKwon .
Could you update the following in `OrcFilterSuite`?
```
 * - OrcFilterSuite uses 'org.apache.orc.storage.ql.io.sarg' package.
 * - HiveOrcFilterSuite uses 'org.apache.hadoop.hive.ql.io.sarg' package.
```

**Comment by HyukjinKwon** [comment]
Sure will update!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (only one minor comment)

**Comment by wangyum** [comment]
Remove [spark-deps-hadoop-2.7-hive-1.2](https://github.com/apache/spark/blob/master/dev/deps/spark-deps-hadoop-2.7-hive-1.2)?

**Comment by dongjoon-hyun** [comment]
Oh, nice catch, @wangyum !

**Comment by HyukjinKwon** [comment]
Nice, thanks @wangyum. Will update.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master for Apache Spark 3.1.0

**Comment by HyukjinKwon** [comment]
Thanks @dongjoon-hyun!

**Pull request closed without merging**